packages/rum-core/src/common/config-service.js

```diff
@@ -257,6 +258,21 @@ class Config {
       })
     }
 
+    const sampleRate = properties.transactionSampleRate
+    if (
+      typeof sampleRate !== 'undefined' &&
+      (typeof sampleRate !== 'number' ||
+        isNaN(sampleRate) ||
```

Comment:
isNaN is not required since we already test for ranges and 'number' type. Am i missing something?

Reply:
I would simply replace this logic with `Number.isFinite(sampleRate)` check and checking the min and max range

Reply:
`isNaN` is actually needed here because of the range check is always fails when the value is `NaN` and type its type is also a number.